feder/monitorings/tests.py

```diff
@@ -262,6 +261,28 @@ def test_template_used(self):
         response = self.client.get(self.get_url())
         self.assertTemplateUsed(response, "monitorings/permission_wizard.html")
 
+    def test_set_permissions(self):
```

Comment:
@ad-m This test fails with traceback:
```
ERROR: test_set_permissions (feder.monitorings.tests.PermissionWizardTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/code/feder/monitorings/tests.py", line 275, in test_set_permissions
    response = self.client.post(self.get_url(), data=data)
  File "/usr/local/lib/python3.8/site-packages/django/test/client.py", line 543, in post
    response = super().post(path, data=data, content_type=content_type, secure=secure, **extra)
  File "/usr/local/lib/python3.8/site-packages/django/test/client.py", line 356, in post
    return self.generic('POST', path, post_data, content_type,
  File "/usr/local/lib/python3.8/site-packages/django/test/client.py", line 422, in generic
    return self.request(**r)
  File "/usr/local/lib/python3.8/site-packages/django/test/client.py", line 503, in request
    raise exc_value
  File "/usr/local/lib/python3.8/site-packages/django/core/handlers/exception.py", line 34, in inner
    response = get_response(request)
  File "/usr/local/lib/python3.8/site-packages/django/core/handlers/base.py", line 115, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/usr/local/lib/python3.8/site-packages/django/core/handlers/base.py", line 113, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/local/lib/python3.8/contextlib.py", line 75, in inner
    return func(*args, **kwds)
  File "/usr/local/lib/python3.8/site-packages/django/views/generic/base.py", line 71, in view
    return self.dispatch(request, *args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/braces/views/_access.py", line 97, in dispatch
    return super(LoginRequiredMixin, self).dispatch(
  File "/usr/local/lib/python3.8/site-packages/formtools/wizard/views.py", line 244, in dispatch
    response = super().dispatch(request, *args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/django/views/generic/base.py", line 97, in dispatch
    return handler(request, *args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/formtools/wizard/views.py", line 305, in post
    return self.render_next_step(form)
  File "/usr/local/lib/python3.8/site-packages/formtools/wizard/views.py", line 316, in render_next_step
    new_form = self.get_form(
  File "/usr/local/lib/python3.8/site-packages/formtools/wizard/views.py", line 411, in get_form
    kwargs = self.get_form_kwargs(step)
  File "/code/feder/monitorings/views.py", line 209, in get_form_kwargs
    user = get_user_model().objects.get(pk=user_pk)
  File "/usr/local/lib/python3.8/site-packages/django/db/models/manager.py", line 82, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/django/db/models/query.py", line 406, in get
    raise self.model.DoesNotExist(
feder.users.models.User.DoesNotExist: User matching query does not exist.
```

Reply:
```
  File "/code/feder/monitorings/tests.py", line 275, in test_set_permissions
    response = self.client.post(self.get_url(), data=data)
...
  File "/code/feder/monitorings/views.py", line 209, in get_form_kwargs
    user = get_user_model().objects.get(pk=user_pk)
```

Dwa twoje fragmenty kodu nie mogą się porozumieć... Zaraz wejdę w to głębiej.

Reply:
Już zszedłem głębiej. Po prostu nadawałeś uprawnienia zupełnie innemu użytkownikowi, bo chwilę wcześniej pobierasz pierwszy znak id użytkownika. Czasem działa, ale nie zawsze. Dobrze, że zapytałem o testy, bo oszczędziliśmy jedną iterację.